content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,136 @@
 slug: 17
 title: 17/WAKU-RLN-RELAY
 name: Waku v2 RLN Relay
-status: raw
+status: draft
 tags: waku-core
 editor: Sanaz Taheri <sanaz@status.im>
 ---
 
-The current specification embodies the details of the spam-protected version of `relay` protocol empowered by Rate Limiting Nullifiers (RLN). 
-<!-- More details on RLN can be found in [this spec]() (TODO: to link the spec). -->
+The `17/WAKU-RLN-RELAY` protocol is an extension of `11/WAKU-RELAY` which additionally provides spam protection using [Rate Limiting Nullifiers (RLN)](/spec/32). 
 
-The security objective is to control the number of PubSub messages that each peer can publish per epoch where epoch is a system design parameter, regardless of the published topic.
+The security objective is to contain spam activity in a GossipSub network by enforcing a global messaging rate to all the peers.
+Peers that violate the messaging rate are considered spammers and their message is considered spam.
+Spammers are also financially punished and removed from the system. 
 
 
 **Protocol identifier***: `/vac/waku/waku-rln-relay/2.0.0-alpha1`
 
 # Motivation
 
-In open p2p messaging networks, one big problem is spam-resistance. 
-Existing solutions, such as Whisper’s proof of work, are insufficient, especially for heterogeneous nodes. 
+In open and anonymous p2p messaging networks, one big problem is spam resistance. 
+Existing solutions, such as Whisper’s proof of work are computationally expensive hence not suitable for resource-limited nodes. 
 Other reputation-based approaches might not be desirable, due to issues around arbitrary exclusion and privacy.
 
-We augment the `relay` protocol with a novel, light, and effective spam prevention mechanism which also suits the resource-constrained nodes.
-
-<!-- TODO: Fill in more -->
+We augment the [`11/WAKU2-RELAY`](/spec/11) protocol with a novel construct of [RLN](/spec/32) to enable an efficient economic spam prevention mechanism that can be run in resource-constrained environments.
 
 
 # Flow
+The messaging rate is set to 1 per `epoch`, 
+where `epoch` indicates the number of `T` seconds elapsed since the Unix epoch event.
```

Comment:
Do we need T, or could we consider making `T` more descriptive? it seems more like a local variable

It seems like we have:

`message-rate` 1
`epoch` <seconds since epoch>

Reply:
`Epoch` is a counter for the multiples of T (`epoch`=`current Unix epoch time`/ `T`), so they are not the same. `T` is the length of each `epoch`. I guess that the naming is confusing, let me know if you have any suggestions for better naming.

Reply:
`epoch=current Unix epoch time/ T` I find this rather confusing, I keep re-reading it but it doesn't make any sense:

- epoch is being redefined _inside_ a definition
- we introduce a generic "T" parameter

Why not do something like:
- unix time epoch is a well understood standard for seconds since 1970
- call T period

Example:

- `unix_time` is 1644810116s
- `period` is 30s
- `epoch is `lceil(unix_time/period)`

This means a message can only be sent every `period`, where `period` is up to the application.